python/tests/catalog/test_glue.py

```diff
@@ -67,7 +67,18 @@ def get_random_databases(n: int) -> Set[str]:
 
 @pytest.fixture(name="_bucket_initialize")
 def fixture_s3_bucket(_s3) -> None:  # type: ignore
-    _s3.create_bucket(Bucket=BUCKET_NAME)
+    bucket = _s3.create_bucket(Bucket=BUCKET_NAME)
+    yield bucket
+
+    response = _s3.list_objects_v2(
+        Bucket=BUCKET_NAME,
+    )
+    while response["KeyCount"] > 0:
+        _s3.delete_objects(Bucket=BUCKET_NAME, Delete={"Objects": [{"Key": obj["Key"]} for obj in response["Contents"]]})
```

Comment:
I haven't checked the details of this test, but delete_objects only accepts 1000 keys at a time for a given bucket. We're probably not creating close to that much here which is why this is passing, but just so the logic is robust we probably want to handle that (if not here, we can do in a separate PR as well)

Reply:
I'm also no expert on the subject, but it looks like the max number of key returned is also 1000:
<img width="776" alt="image" src="https://user-images.githubusercontent.com/1134248/215590530-0511132d-cc2a-474f-acbe-59b70bb8f736.png">
From: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.list_objects_v2

This will iterate until `KeyCount` is zero